src/rpc/client.rs

```diff
@@ -356,7 +356,8 @@ impl ClientAsync {
                     .await;
                 match send_result {
                     Ok(_) => {
-                        let result = self.queue.wait_for_callback(id);
+                        let result =
+                            tokio::task::block_in_place(move || self.queue.wait_for_callback(id));
```

Comment:
better to add a comment about future plans.

Reply:
done